neo/rawio/openephysbinaryrawio.py

```diff
@@ -192,10 +224,44 @@ def _parse_header(self):
                             "SYNC channel is not present in the recording. " "Set load_sync_channel to False"
                         )
 
-                    if has_sync_trace and not self.load_sync_channel:
-                        self._stream_buffer_slice[stream_id] = slice(None, -1)
+                    # Check if ADC and non-ADC channels are contiguous
+                    is_channel_adc = ["ADC" in ch["channel_name"] for ch in info["channels"]]
+                    first_adc_index = is_channel_adc.index(True) if any(is_channel_adc) else len(is_channel_adc)
+                    non_adc_channels_after_adc_channels = [not flag for flag in is_channel_adc[first_adc_index:]]
+                    if any(non_adc_channels_after_adc_channels):
+                        raise ValueError(
+                            "Interleaved ADC and non-ADC channels are not supported. "
+                            "ADC channels must be contiguous. Open an issue in python-neo to request this feature."
+                        )
```

Comment:
I think this would be easier to read:
```
                if any(is_channel_adc):
                    first_adc_index = is_channel_adc.index(True)
                    non_adc_channels_after_adc_channels = [not is_adc for is_adc in is_channel_adc[first_adc_index:]]
                    if any(non_adc_channels_after_adc_channels):
                        raise ValueError(
                            "Interleaved ADC and non-ADC channels are not supported. "
                            "ADC channels must be contiguous. Open an issue in python-neo to request this feature."
                        )
```
I found it a bit hard to understand the logic if no ADC channels are present (then `non_adc_channels_after_adc_channels` would be an empty list)

Reply:
Yes, this is better.